src/components/Layout/Header.component.test.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { render, screen } from '@testing-library/react';
+import Header from './Header.component';
+
+describe('Header Component', () => {
+
+  it('renders correctly', () => {
+    const tree = renderer.create(<Header/>).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+  it('Title is correct', () => {
+    render(<Header/>);
+    const i = screen.getByText(/Dark Mode/i);
+    expect(i).toBeInTheDocument();
+  });
```

Comment:
You could test the functionality of typing in the search bar as well and checking that the component has the text that you "wrote" in it.

Reply:
I didn't think of that, it sounds good.